common/thorhelper/thorcommon.cpp

```diff
@@ -2123,6 +2123,15 @@ static bool getTranslators(Owned<const IDynamicTransform> &translator, Owned<con
             if (publishedCrc && expectedCrc && (publishedCrc != expectedCrc))
                 DBGLOG("Overriding stored record layout reading file %s", tracing);
         }
+        else if (mode == RecordTranslationMode::PayloadRemoveOnly)
+        {
+            if (publishedCrc && expectedCrc && (publishedCrc != expectedCrc))
+            {
+                DBGLOG("Overriding stored record layout reading file %s", tracing);
+                if (expectedFormat->queryRecordAccessor(true).getNumFields() > publishedFormat->queryRecordAccessor(true).getNumFields())
```

Comment:
I'm not sure if this is supposed to check that fields were only removed and not added, but how would it happen a case where fields were both added and removed (but more were removed than added)?

Reply:
I fixed the check to use match_none. Now it strictly checks for added fields.